clusterloader2/pkg/measurement/common/api_availability_measurement.go

```diff
@@ -48,43 +48,85 @@ func createAPIAvailabilityMeasurement() measurement.Measurement {
 	stopCh              chan struct{}
-	hosts               []string
+	frequency           time.Duration
+	masterIPs           []string
 	summaries           []measurement.Summary
 	clusterLevelMetrics *apiAvailabilityMetrics
-	hostLevelMetrics    map[string]*apiAvailabilityMetrics
+	masterLevelMetrics  map[string]*apiAvailabilityMetrics
 	wg                  sync.WaitGroup
 }
 
-func (a *apiAvailabilityMeasurement) updateMasterAvailabilityMetrics(c clientset.Interface, config *measurement.Config, provider provider.Provider) {
-	for _, host := range a.hosts {
-		// SSH and check the health of the host
-		command := fmt.Sprintf("curl -f -s -k %slocalhost:%v/healthz", "https://", 443)
-		sshResult, err := measurementutil.SSH(command, host+":22", provider)
-		availability := err != nil || sshResult.Code != 0
-		a.updateAvailabilityMetrics(availability, a.hostLevelMetrics[host])
+func (a *apiAvailabilityMeasurement) updateMasterAvailabilityMetrics(c clientset.Interface, provider provider.Provider) {
+	for _, ip := range a.masterIPs {
+		statusCode, err := a.pingMaster(ip)
+		availability := statusCode == strconv.Itoa(http.StatusOK)
+		if err != nil {
+			klog.Warningf("execservice issue: %w", err)
+			availability = false
+		}
+		a.masterLevelMetrics[ip].update(availability)
 	}
 }
 
+func (a *apiAvailabilityMeasurement) pingMaster(masterIP string) (string, error) {
+	pod, err := execservice.GetPod()
+	if err != nil {
+		return "", fmt.Errorf("problem with GetPod(): %w", err)
+	}
+	cmd := fmt.Sprintf("curl --connect-timeout 5 -s -k -w \"%%{http_code}\" -o /dev/null https://%s:443/healthz", masterIP)
+	output, err := execservice.RunCommand(pod, cmd)
+	if err != nil {
+		return "", fmt.Errorf("problem with RunCommand(): %w", err)
+	}
+	return output, nil
+}
+
 func (a *apiAvailabilityMeasurement) updateClusterAvailabilityMetrics(c clientset.Interface) {
-	// Check the availability of the cluster by issuing a REST call to /healthz end point
-	result := c.CoreV1().RESTClient().Get().AbsPath("/healthz").Do(context.TODO())
+	result := c.CoreV1().RESTClient().Get().AbsPath("/livez").Do(context.TODO())
 	status := 0
 	result.StatusCode(&status)
-	a.updateAvailabilityMetrics(status == 200, a.clusterLevelMetrics)
+	availability := status == http.StatusOK
+	if !availability {
+		klog.Warning("cluster not available")
+	}
+	a.clusterLevelMetrics.update(availability)
 }
 
-func (a *apiAvailabilityMeasurement) start(config *measurement.Config, SSHToMasterSupported bool, probeDuration int) error {
-	a.hosts = config.ClusterFramework.GetClusterConfig().MasterIPs
-	if len(a.hosts) < 1 {
-		return goerrors.Errorf("APIAvailability measurement can't start due to lack of master IPs")
+func (a *apiAvailabilityMeasurement) Execute(config *measurement.Config) ([]measurement.Summary, error) {
+	action, err := util.GetString(config.Params, "action")
+	if err != nil {
+		return nil, err
 	}
 
+	switch action {
+	case "start":
+		return nil, a.start(config)
+	case "hold":
```

Comment:
Maybe pause? And unpause? Same for the variable names.

Reply:
Agree - pause sounds more intuitive